haystack/document_stores/base.py

```diff
@@ -191,22 +191,27 @@ def get_document_by_id(self, id: str, index: Optional[str] = None) -> Optional[D
     def get_document_count(self, filters: Optional[Dict[str, List[str]]] = None, index: Optional[str] = None) -> int:
         pass
 
+    @staticmethod
     @njit#(fastmath=True)
-    def normalize_embedding(self, emb: np.ndarray) -> None:
+    def normalize_embedding(emb: np.ndarray) -> None:
         """
-        Performs L2 normalization of embeddings vector inplace. Input can be a single vector (1D array) or a matrix (2D array).
+        Performs L2 normalization of embeddings vector inplace. Input can be a single vector (1D array) or a matrix
+        (2D array).
         """
         # Might be extended to other normalizations in future
 
         # Single vec
         if len(emb.shape) == 1:
-            norm = np.sqrt(emb.dot(emb)) #faster than np.linalg.norm()
+            norm = np.sqrt(emb.dot(emb))  # faster than np.linalg.norm()
             if norm != 0.0:
                 emb /= norm
         # 2D matrix
         else:
-            norm = np.linalg.norm(emb, axis=1)
-            emb /= norm[:, None]
+            for vec in emb:
+                vec = np.ascontiguousarray(vec)
+                norm = np.sqrt(vec.dot(vec))
+                if norm != 0.0:
+                    vec /= norm
 
```

Comment:
Can you please do a quick check that this for loop is still faster than the plain numpy normalization via linalg.norm? Speed was the only reason for introducing it. If it's now slower, we should probably get rid of it.

Reply:
I just checked execution time, when having numba installed, the for-loop is faster than `np.linalg.norm`, otherwise `np.linalg.norm` would be faster.

Reply:
ok then let's stick with numba